src/components/sharygin-cubic-graph.js

```diff
@@ -18,8 +18,16 @@ const getScaledPointsList = pointsDict => {
 };
 
 const getLongLine = (x1, y1, x2, y2) => {
-  const dx = x1 - x2;
-  const dy = y1 - y2;
+  let dx, dy;
+  if (x1 !== x2) {
+    dx = x1 - x2;
+    dy = y1 - y2;
+  }
+  else {
+    const c = 1 - (x1 + y1) * (x1 + y1 + 1)
+    dx = c - y1 * (2 * y1 + 1)
+    dy = c + x1 * (2 * x1 + 1)
```

Comment:
```suggestion
    dy = -c + x1 * (2 * x1 + 1)
```
Tiny sign error here I think.